tests/shouldwork/Cores/Xilinx/DcFifo/Abstract.hs

```diff
@@ -122,34 +122,26 @@ mkTestBench cFifo = done
  where
   (rClk, wClk) = biTbClockGen (not <$> done)
 
-  rRst = resetGen
-  wRst = resetGen
+  noRRst = unsafeFromHighPolarity $ pure False
```

Comment:
Is this just to make it simpler? The tests still pass with `resetGen`!

Reply:
The product guide specifically says that you cannot push data to the FIFO or pop data from the FIFO until both sides have entered and left reset. We don't enforce that in our test. It might be that it still passes, but it is undefined behaviour nonetheless.

For the test in `DcFifo/Basic.hs`, you could say that we don't break that rule, but I think it's better to just leave the whole issue of reset well alone in these tests and write a separate test where we confirm that our model matches Vivado's model if we observe the proper reset sequence. We don't have to do anything about the improper reset sequence. Although it might be a nice feature if we actually errored in Haskell simulation if someone would simulate an improper reset sequence.

Reply:
Fair enough, this is definitely easier to think about :)

Thanks for the explanation!